frontend/amundsen_application/static/js/pages/LineagePage/index.tsx

```diff
@@ -0,0 +1,217 @@
+
+import { GlobalState } from 'ducks/rootReducer';
+import Breadcrumb from 'components/Breadcrumb';
+
+import LineageGraph from 'components/LineageGraph';
+
+import './styles.scss';
+
+import { getLink } from 'components/ResourceListItem/TableListItem';
+import * as Constants from './constants';
+
+const SERVER_ERROR_CODE = 500;
+
+export interface PropsFromState {
+  isLoading: boolean;
+  statusCode: number | null;
+  lineageTree: Lineage;
+}
+export interface DispatchFromProps {
+  getTableLineageDispatch: (
+    key: string,
+    depth?: number,
+    direction?: string
+  ) => GetTableLineageRequest;
+  getColumnLineageDispatch: (
+    key: string,
+    columnName: string,
+    depth?: number,
+    direction?: string
+  ) => GetColumnLineageRequest;
+}
+
+export interface MatchProps {
+  cluster: string;
+  database: string;
+  schema: string;
+  table: string;
+}
+
+export type LineageProps = PropsFromState &
+  DispatchFromProps &
+  RouteComponentProps<MatchProps>;
+
+const ErrorMessage = () => (
+  <div className="container error-label">
+    <Breadcrumb />
+    <label>{Constants.ERROR_MESSAGE}</label>
+  </div>
+);
+
+export class LineagePage extends React.Component<
+  LineageProps & RouteComponentProps<any>
+> {
+  state: any = {};
+
+  componentDidMount() {
+    const { getTableLineageDispatch } = this.props;
+    this.key = this.getTableKey();
+    getTableLineageDispatch(this.key, 5);
+    this.didComponentMount = true;
+  }
+
+  componentDidUpdate() {
+    const { getTableLineageDispatch } = this.props;
+    const newKey = this.getTableKey();
+    if (this.key !== newKey) {
+      this.key = newKey;
+      getTableLineageDispatch(this.key, 5);
+    }
+  }
+
+  private didComponentMount: boolean = false;
+
+  private key: string;
+
+  getDisplayName() {
+    const { match } = this.props;
+    const { params } = match;
+
+    return `Lineage Information | ${params.schema}.${params.table}`;
+  }
+
+  // ToDo: Should be moved to a common place
+  getTableKey() {
+    /*
+    This 'key' is the `table_uri` format described in metadataservice. Because it contains the '/' character,
+    we can't pass it as a single URL parameter without encodeURIComponent which makes ugly URLs.
+    DO NOT CHANGE
+    */
+    const { match } = this.props;
+    const { params } = match;
+
+    return `${params.database}://${params.cluster}.${params.schema}/${params.table}`;
+  }
+
+  render() {
+    const { match, lineageTree, statusCode, isLoading } = this.props;
+    const { params } = match;
+    let innerContent;
+    // We want to avoid rendering the previous table's metadata before new data is fetched in componentDidMount
+    if (isLoading || !this.didComponentMount) {
+      innerContent = (
+        <div className="pulse-lineage">
+          <div className="pulse-lineage-media is-pulse-animated" />
+        </div>
+      );
+    } else if (statusCode === SERVER_ERROR_CODE) {
+      innerContent = <ErrorMessage />;
+    } else {
```

Comment:
I would extract everything on this branch into a new component and leave this file as a simple branching component with the connection to the redux store.

This will allow to test it a lot easier.